hail/python/test/hail/utils/test_utils.py

```diff
@@ -170,7 +170,7 @@ def test_hadoop_ls_file_that_does_not_exist(self):
         except FileNotFoundError:
             pass
         except FatalError as err:
-            assert 'FileNotFoundException: a_file_that_does_not_exist' in err.args[0]
+            assert 'FileNotFoundException: file:/io/a_file_that_does_not_exist' in err.args[0]
```

Comment:
I admittedly was getting frustrated with Hadoop at this point. I could change the HadoopFSURL.toString to elide `file:`, but that would be confusing when you're actually in an HDFS cluster. On the other hand, including file: is confusing if you're not in an HDFS cluster.


On the specific question of `file:` versus `file:/`, `file:` is the correct choice. The [URL grammar](https://en.wikipedia.org/wiki/URL#/media/File:URI_syntax_diagram.svg) calls the stuff before the `:` the "scheme". The reason why web URLs are always `http://` is because they have a "host" portion (the domain name). File systems don't have a host because they're assumed to be local. The `/` in `file:/` is just the name of the root directory of the file system: `/`.